src/Components/Web/src/Forms/InputFile.cs

```diff
@@ -82,9 +79,12 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
         }
 
         internal Stream OpenReadStream(BrowserFile file, long maxAllowedSize, CancellationToken cancellationToken)
-            => _jsUnmarshalledRuntime != null ?
-                new SharedBrowserFileStream(JSRuntime, _jsUnmarshalledRuntime, _inputFileElement, file) :
-                new RemoteBrowserFileStream(JSRuntime, _inputFileElement, file, Options.Value, maxAllowedSize, cancellationToken);
+            => new BrowserFileStream(
+                JSRuntime,
+                _inputFileElement,
+                file,
+                _jsUnmarshalledRuntime == null ? maxAllowedSize : long.MaxValue, // Don't enforce the maxAllowedSize for WASM (to preserve legacy behavior).
+                cancellationToken);
```

Comment:
Do you know if there's an E2E test showing that the `maxAllowedSize` is enforced on server? Would be ideal to have one now there's conditional logic here (in case some time in the future `_jsUnmarshalledRuntime` somehow becomes nonnull on Server due to some unrelated change).

Reply:
So I looked into this, and it appears we already had that test:

https://github.com/dotnet/aspnetcore/blob/1395fa8574a574ca969463952d4c5e89e9febe89/src/Components/test/E2ETest/Tests/InputFileTest.cs#L186-L203

So why was `maxAllowedSize` being enforced? Because we had a higher level check (with the associated InputFile specific error message):

https://github.com/dotnet/aspnetcore/blob/1395fa8574a574ca969463952d4c5e89e9febe89/src/Components/Web/src/Forms/InputFile/BrowserFile.cs#L42-L45

Backing out this special casing.